### ISSUE TITLE
Leakage? 

### ISSUE THREAD

**Issue by onacrame** [KurochkinAlexey/Self-boosted-Time-series-Forecasting#1]
EEMD is run on the full dataset and then subsetting of train and test is done after.  Appreciate the min max scaling for the training set is applied to the test set but is not using EEMD features computed at the outset introducing leakage?  What am I missing?

**Comment by onacrame** [comment]
Even with respect to the training set shouldn’t EEMD features computed for every data point up to and including the current training sample.  It looks like what this is doing is computing EEMD features for the whole series to begin with.  I’d assume EEMD features would be different if they were computed up to and including the training sample compared to the same features computed on the whole series but then subsetted for any given training sample.

**Comment by q-learning-trader** [comment]
Totally agree with you @MotoRZR 
I believe this is a case of leakage.
It would be nice if the author @KurochkinAlexey could reply.
Thanks
